### PR TITLE
Adds service account authentication to GCE fence agent

### PR DESCRIPTION
**Pull request by megelatim**
Adds the ability to use a service acount for authentiation using --serviceaccount.
Adds additional debug information.

### CODE REVIEW

**Comment by wenningerk** [comment]
test this please

**Comment by oalbrigt** [comment]
We need to fall back to the old solution when running with old libraries for backwards compatibility.

**Comment by megelatim** [comment]
> test this please

Tested - had to add oauth2client to the travis pip installs

> We need to fall back to the old solution when running with old libraries for backwards compatibility.

This was tested on python 2.7 and python 3 (RHEL 7 & 8 & SLES 12 & 15)

**Comment by oalbrigt** [comment]
ok to test

File: agents/gce/fence_gce.py [hunk 1]
```diff
@@ -20,6 +28,8 @@
 
 import googleapiclient.discovery
 from fencing import fail_usage, run_delay, all_opt, atexit_handler, check_input, process_input, show_docs, fence_action
+from oauth2client.client import GoogleCredentials
```
Comment (oalbrigt) [inline agents/gce/fence_gce.py:31]: Can do do these in a try/except: pass ?

That should make it able to be built (e.g. manpage generation) even on FreeBSD, where we dont have oauth2client currently installed.

Comment (megelatim) [reply agents/gce/fence_gce.py]: Added the try/except:pass to the oauth2client.* imports

**Comment by oalbrigt** [comment]
LGTM. Thanks.

**Pull request merged**